File: src/python/log_injection.py
"""CWE-117: Log injection.

Rule: py/log-injection
Autofix: typically NOT provided — fix requires a sanitizer (strip CR/LF,
encode) and possibly switching to structured logging.
"""

import logging
from flask import Flask, request

logging.basicConfig(level=logging.INFO)
log = logging.getLogger(__name__)

app = Flask(__name__)


@app.route("/login")
def login():
    user = request.args.get("user", "")
    log.info("login attempt by " + user)
    return "ok"


@app.route("/audit")
def audit():
    action = request.args.get("action", "")
    log.warning(f"[audit] action={action}")
    return "ok"


if __name__ == "__main__":
    app.run(port=4109)

File: src/javascript/log-injection.js
// CWE-117: Log injection (CRLF, log forging).
// Rule: js/log-injection
// Autofix: typically NOT provided — fix requires choosing a sanitizer
// (e.g. stripping CR/LF, encoding) and potentially a structured-log lib.

const express = require('express');
const app = express();

app.get('/login', (req, res) => {
  const user = String(req.query.user || '');
  console.log('login attempt by ' + user);
  res.send('ok');
});

app.get('/audit', (req, res) => {
  const action = String(req.query.action || '');
  console.warn(`[audit] action=${action}`);
  res.send('ok');
});

app.listen(3112);
